### DISCUSSION TITLE
Any way to get the runtime dir/plugin path?

### DISCUSSION THREAD

**Discussion by MLFlexer** [wezterm/wezterm#3989] [Q&A]
Hi,
I'm trying to develop at plugin which requires a path to a shell script, which can be installed via. git by the experimental plugin require:  [https://github.com/wez/wezterm/commit/e4ae8a844d8feaa43e1de34c5cc8b4f07ce525dd](https://github.com/wez/wezterm/commit/e4ae8a844d8feaa43e1de34c5cc8b4f07ce525dd)

I would like to reference this shell script, but I cannot find a good way of getting the path to the runtime dir or plugin dir. Any ideas?

**Comment by nekowinston** [comment]
Nice to see someone else mess around with the Plugin API!

I don't think we can do that at the moment, without making assumptions in the code.
Technically, the info is there in `package.path` and `wezterm.plugin.list()`:
`$XDG_RUNTIME_DIR` is used if available to [construct the runtime directory](https://github.com/wez/wezterm/blob/115e629ec14757430462ce4d0ba8e8bd4e193fab/config/src/lua.rs#L234-L237) for `package.path`, e.g. `/run/user/1000/wezterm/plugins/?/plugin/init.lua`, or [falls back](https://github.com/wez/wezterm/blob/115e629ec14757430462ce4d0ba8e8bd4e193fab/config/src/config.rs#L1646-L1652) to `~/.local/share/wezterm`.

And `wezterm.plugin.list()`only shows the resolved directory name relative to that plugin path. I might be overlooking some lua function here, but I *think* that we're not able to get the path to the plugin reliably at the moment.

**Reply by MLFlexer** [reply]
Yea, that's what I suspected. Would be a nice feature to have eventually

**Comment by wez** [comment]
Would adding a `plugin_dir` field to the `RepoSpec` which is returned by `wezterm.plugin.list()` do what you're looking for? If so, please file an enhancement issue for that!

**Reply by MLFlexer** [reply]
yes, that would probably be a good place to add it. ~~Will file the enhancement~~ 👍  [https://github.com/wez/wezterm/issues/4015](https://github.com/wez/wezterm/issues/4015)
